scheds/rust/scx_lavd/src/bpf/main.bpf.c

```diff
@@ -1956,26 +1983,49 @@ static void put_global_rq(struct task_struct *p, struct task_ctx *taskc,
 	calc_when_to_run(p, taskc, cpuc, enq_flags);
 
 	/*
-	 * Try to find and kick a victim CPU, which runs a less urgent task.
-	 * The kick will be done asynchronously.
+	 * If a task is eligible, dispatch to the eligible DSQ.
 	 */
-	try_find_and_kick_victim_cpu(p->cpus_ptr, taskc);
+	if (is_eligible(taskc)) {
+		/*
+		 * Try to find and kick a victim CPU, which runs a less urgent
+		 * task. The kick will be done asynchronously.
+		 */
+		try_find_and_kick_victim_cpu(p->cpus_ptr, taskc);
+
+		/*
+		 * If the current task has something to yield, try preempt it.
+		 */
+		p_run = bpf_get_current_task_btf();
+		taskc_run = try_get_task_ctx(p_run);
+		if (taskc_run && p_run->scx.slice != 0)
+			try_yield_current_cpu(p_run, cpuc, taskc_run);
+
+		goto dispatch_out;
+	}
 
 	/*
-	 * If the current task has something to yield, try preempt it.
+	 * If the task is ineligible, keep it to the ineligible run queue.
 	 */
-	p_run = bpf_get_current_task_btf();
-	taskc_run = try_get_task_ctx(p_run);
-	if (taskc_run && p_run->scx.slice != 0)
-		try_yield_current_cpu(p_run, cpuc, taskc_run);
+	t = bpf_obj_new(typeof(*t));
```

Comment:
I think it should be possible to pre-allocate this node on `task_init()` and keep it on taskc so that enqueue path doesn't have to do dynamic allocations but at the same time `bpf_obj_new()` might be cheap enough for this to not matter.

Reply:
nvm, this gets removed later.